src/services/invitation/index.ts

```diff
@@ -1,16 +1,80 @@
+import { group } from 'console';
 import { StatusCodes } from 'http-status-codes';
+import * as Papa from 'papaparse';
+import { parse, resolve } from 'path';
+import { Int32 } from 'typeorm';
+import { v4 as uuidv4 } from 'uuid';
 
+import { BusboyFileStream } from '@fastify/busboy';
+import fastifyMultipart, { MultipartFile } from '@fastify/multipart';
 import { FastifyPluginAsync } from 'fastify';
 import fp from 'fastify-plugin';
 
-import { IdParam } from '@graasp/sdk';
+import { HttpMethod, IdParam, PermissionLevel } from '@graasp/sdk';
+import {
+  FolderItemType,
+  ItemSettings,
+  ItemType,
+  UnknownExtra,
+  buildPathFromIds,
+} from '@graasp/sdk';
 
 import { Repositories, buildRepositories } from '../../utils/repositories';
+import { validatePermission } from '../authorization';
+import { Item, ItemExtra } from '../item/entities/Item';
+import { item } from '../item/fluent-schema';
+import { upload } from '../item/plugins/file/schema';
+import { ItemService } from '../item/service';
 import { Actor, Member } from '../member/entities/member';
+import serviceApi from '../websockets/service-api';
 import { Invitation } from './invitation';
 import definitions, { deleteOne, getById, getForItem, invite, sendOne, updateOne } from './schema';
 import { InvitationService } from './service';
 
+type CSVInvite = {
+  email: string;
+  name?: string;
+  group_name?: string;
+  permission?: PermissionLevel;
+};
+
+function createNewFolder(member: Actor, rowInvite: CSVInvite, parentId: string): Promise<Item> {
+  return new Promise((resolve) => {
+    const buildId = uuidv4();
+    let buildPath = buildPathFromIds(buildId);
+    if (parentId) buildPath = `${parentId}.${buildPath}`;
```

Comment:
I guess you wouldn't need to specify the path if the uuid is provided by the item service. See my comment above.